node/src/actors/chain_manager/mining.rs

```diff
@@ -863,6 +983,36 @@ fn build_block(
     (block_header, txns)
 }
 
+/// Function that return an Option of a SuperBlock or None in case if there are not blocks
+/// to produce a SuperBlock
+fn build_superblock(
+    block_headers: &[BlockHeader],
+    sorted_ars_identities: &[PublicKeyHash],
+    index: u32,
+    last_block_in_previous_superblock: Hash,
+) -> Option<SuperBlock> {
+    let last_block = block_headers.last()?.hash();
+    let merkle_drs: Vec<Hash> = block_headers
+        .iter()
+        .map(|b| b.merkle_roots.dr_hash_merkle_root)
+        .collect();
+    let merkle_tallies: Vec<Hash> = block_headers
+        .iter()
+        .map(|b| b.merkle_roots.tally_hash_merkle_root)
+        .collect();
+
+    let pkh_hashes: Vec<Hash> = sorted_ars_identities.iter().map(|pkh| pkh.hash()).collect();
```

Comment:
Hashes of the public key hashes? :open_mouth:

Reply:
currently our pkh is not a really hash, it is only the first part so... hashing the hashes :P 
The other solution should be to have a real PublicKeyHash, why do we reduce them before?

Reply:
Anyway, now we are creating a merkleroot with our pkh, but I remeber that it would be with the BLS addresses, that it will be done in the issue #1236

Reply:
It's OK for the time being.

And yes, we should stop trimming PKHs asap.